certbot/tests/crypto_util_test.py

```diff
@@ -236,6 +239,80 @@ def test_it(self):  # pylint: disable=no-self-use
+
+    def setUp(self):
+        self.renewable_cert = mock.MagicMock()
+        self.renewable_cert.chain = SS_CERT_PATH
+        self.renewable_cert.cert = SS_CERT_PATH
+        self.renewable_cert.privkey = RSA512_KEY_PATH
+        self.renewable_cert.fullchain = test_util.vector_path('self_signed_fullchain.pem')
+
+    @classmethod
+    def _verify_fullchain(cls, renewable_cert):
+        from certbot.crypto_util import verify_fullchain
+        return verify_fullchain(renewable_cert)
+
+    @classmethod
+    def _verify_cert_matches_priv_key(cls, renewable_cert):
+        from certbot.crypto_util import verify_cert_matches_priv_key
+        return verify_cert_matches_priv_key(renewable_cert)
+
+    @classmethod
+    def _verify_renewable_cert_sig(cls, renewable_cert):
+        from certbot.crypto_util import verify_renewable_cert_sig
+        return verify_renewable_cert_sig(renewable_cert)
+
+    @classmethod
+    def _verify_renewable_cert(cls, renewable_cert):
+        from certbot.crypto_util import verify_renewable_cert
+        return verify_renewable_cert(renewable_cert)
+
+    def test_fullchain_matches(self):
+        self.assertEqual(None, self._verify_fullchain(self.renewable_cert))
+
+    def test_fullchain_mismatch(self):
+        bad_renewable_cert = mock.MagicMock()
+        bad_renewable_cert.chain = SS_CERT_PATH
+        bad_renewable_cert.cert = SS_CERT_PATH
+        bad_renewable_cert.fullchain = SS_CERT_PATH
+
+        self.assertRaises(AssertionError, self._verify_fullchain, bad_renewable_cert)
+
+    def test_fullchain_ioerror(self):
+        bad_renewable_cert = mock.MagicMock()
+        bad_renewable_cert.chain = "dog"
+        self.assertRaises(IOError, self._verify_fullchain, bad_renewable_cert)
+
+    def test_cert_priv_key_match(self):
+        self.assertEqual(None, self._verify_cert_matches_priv_key(self.renewable_cert))
+
+    def test_cert_priv_key_mismatch(self):
+        bad_renewable_cert = mock.MagicMock()
+        bad_renewable_cert.privkey = RSA256_KEY_PATH
+        bad_renewable_cert.cert = SS_CERT_PATH
+
+        self.assertRaises(OpenSSL.SSL.Error, self._verify_cert_matches_priv_key, bad_renewable_cert)
+
+    def test_cert_sig_match(self):
+        self.assertEqual(None, self._verify_renewable_cert_sig(self.renewable_cert))
+
+    def test_cert_sig_mismatch(self):
+        bad_renewable_cert = mock.MagicMock()
+        bad_renewable_cert.chain = SS_CERT_PATH
+        bad_renewable_cert.cert = test_util.vector_path('self_signed_cert_bad.pem')
+
+        self.assertRaises(OpenSSL.crypto.Error, self._verify_renewable_cert_sig, bad_renewable_cert)
+
+    def test_verify_renewable_cert(self):
+        self.assertEqual(None, self._verify_renewable_cert(self.renewable_cert))
+
+    @mock.patch('certbot.crypto_util.verify_renewable_cert_sig', side_effect=errors.Error(""))
+    def test_verify_renewable_cert_failure(self, mock_verify_renewable_cert_sign):
+        #pylint: disable=unused-argument
```

Comment:
If you want, you can remove the need for this line by prepending `unused_` to the beginning of the mock argument. Either way is fine though.